test/sanity/code-smell/changelog.py

```diff
@@ -18,6 +18,9 @@ def main():
         if ext not in allowed_extensions:
             print('%s:%d:%d: extension must be one of: %s' % (path, 0, 0, ', '.join(allowed_extensions)))
 
+        if os.path.basename(path).startswith('.'):
+            print('%s:%d:%d: file must not be a dotfile' % (path, 0, 0))
```

Comment:
Shouldn't it be a `continue` instead of a `print()`, and happen before the extension check? This way, a file like `.keep` is flagged as an error, instead of being ignored (what the changelog tool does).

Reply:
Compare `devel` (`stable-2.10` should be the same): https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_data/sanity/code-smell/changelog.py#L23-L29

Reply:
The intent is to have the sanity check report an error for any file in the fragments directory that isn't processed by the changelog tool. That way an incorrectly named file doesn't get accidentally omitted from the changelog.

Reply:
Wouldn't that be the wrong behavior for `.keep` or `.gitkeep` files? Or do we have to add ignore.txt entries for them?

Reply:
The implementation in 2.9 makes no allowance for "keep" files. If that's something we want in devel we'll need to make sure that the list of special "keep" files is the same for both the changelog tool and the sanity test.

Reply:
I see the behavior in `antsibull-changelog` is to ignore all dot files. For the sanity test (in devel) we could allow just `.keep` and error on other dot files, since there shouldn't be stray files in the changelog directory anyways.